client/state/data-layer/wpcom-api-middleware.js

```diff
@@ -0,0 +1,21 @@
+/**
+ * Internal dependencies
+ */
+import handlers from './wpcom';
+
+/**
+ * WPCOM Middleware API
+ *
+ * Intercepts actions requesting data provided by the
+ * WordPress.com API and passes them off to the
+ * appropriate handler.
+ */
+export const middleware = store => next => action =>
+	// we won't use has( handlers, action.type )
+	// here becuase of performance implications
```

Comment:
I think it'll be interesting to keep in mind in the future though, since I wonder if we'll be able to expect better lookup performance from native `Map` than objects.

In trying to look for some resources, I found an interesting tidbit from [this comment](http://stackoverflow.com/a/18541990), which is that you can create an object without prototype members via `Object.create( null )`. Benchmarks I've seen show that `in` and `'undefined' === typeof` checks tend to perform better than `Object#hasOwnProperty` but are susceptible to issues of looking up prototype keys (`valueOf`, `toString`).

So, while the current implementation here is _fast_, it is also susceptible to passing an action of `{ type: 'valueOf' }` (try it!). To fix this, you could revise `mergeHandlers` to create the object via `Object.create` to have both the performance benefit and lack of prototype member edge cases.

```js
const handlers = Object.create( null );
handlers[ PLANS_REQUEST ] = requestPlans;

export const middleware = ( store ) => ( next ) => ( action ) =>
	!! handlers[ action.type ] ?
		// ...
```